### DISCUSSION TITLE
[3.7.0-beta] Make it possible to opt out of automatic merging into drafts

### DISCUSSION THREAD

**Discussion by wsydney76** [craftcms/cms#9053] [Ideas] [answered]
### Description

> Changes from an entry’s Current revision are now automatically merged into drafts upon visiting drafts’ edit pages.

I appreciate every step towards supporting collaborative work, but everything with the word 'automatic' in it lets the bells ring, especially now that provisional drafts are unvisible to others. It is likely to produce unwanted results, that can even go unnoticed.

For example, a draft that has been checked and accepted in all details by colleagues can be changed by provisional fixes in the current revision.

Right now we're just adding some code to make conflicting updates impossible, which fits our workflow but may not be a general solution.

So as long as there is no UI for comparing/accepting/rejecting changes I suggest to make it possible to opt out of automatic merging.

Looking into the code I saw there is a EVENT_BEFORE_MERGE_CANONICAL_CHANGES event, may be the simplest thing for now is making this event cancellable?

**Comment by brandonkelly** [comment, marked as answer]
> I appreciate every step towards supporting collaborative work, but everything with the word 'automatic' in it lets the bells ring, especially now that provisional drafts are unvisible to others. It is likely to produce unwanted results, that can even go unnoticed.

Prior to 3.7, upstream changes were still merged into a draft automatically when you clicked **Publish draft**, regardless of whether you had opted into doing it while editing the draft. So it’s not a new behavior. The only change in 3.7 was that you no longer have to remember manually press the “Merge changes” button to see what the result of that merge is going to be, before you apply the draft. So I would argue that it’s actually _less_ likely for these merged changes to go unnoticed in 3.7.

> So as long as there is no UI for comparing/accepting/rejecting changes I suggest to make it possible to opt out of automatic merging.

This should be resolved when we get to adding a content comparison view (#8135). At that point we can make it possible to see all of the upstream changes and manually choose which ones should be merged into the draft.

Until then, I don’t have any interest in making the current behavior configurable, as there is no other way to propagate upstream changes to drafts. And I’m sure it would be extremely confusing/annoying to authors if they had to constantly recreate the same content changes because they never got merged into drafts.

> Right now we're just adding some code to make conflicting updates impossible

Whenever upstream changes are merged into a draft, Craft is already ignoring any fields that have already been changed within the draft. So you don’t really have to worry about conflicting updates, if I’m understanding you correctly.
